### PR TITLE
[TR3FIR-141] Update embassy to latest

### PR DESCRIPTION
**Pull request by mattcaron-token**
There are 3 merges here which likely don't need to be fully reviewed:

1. Merge of `main` into this branch (at the time, it was named `pull_upstream2`). This is just upstream commits, **but** I do resolve some conflicts. The conflict resolution bears review.
2. Merge of `token-main` into this branch. This is commits from Taylor's PR which have already been reviewed.
3. Another merge of `main` into this branch. This is also just upstream commits.

There are also 3 commits which are normal commits.

Everything that should be reviewed is tagged with [TR3FIR-141]. Everything else is just nonconflicting merges, etc.

This also implements the new versioning scheme.

[TR3FIR-141]: https://tokenring.atlassian.net/browse/TR3FIR-141?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: embassy-stm32/src/time_driver.rs [hunk 1]
```diff
@@ -367,6 +362,11 @@ impl RtcDriver {
                 self.next_period();
             }
 
+            // Half overflow
+            if sr.ccif(0) {
+                self.next_period();
+            }
+
```
Comment (TaylorCarpenter) [inline embassy-stm32/src/time_driver.rs:369]: This will cause the tick rate to be double-time.

**Review by TaylorCarpenter — CHANGES_REQUESTED** [review]

File: embassy-stm32/src/time_driver.rs (hunk 1 above)
Comment (mattcaron-token) [reply embassy-stm32/src/time_driver.rs]: You are correct.

I saw you removed it and noted that change when it conflicted, but it looks like I accepted the wrong side.

Pushed a commit to fix this.

**Review by TaylorCarpenter — APPROVED** [review]

**Comment by mattcaron-token** [comment]
I'm not merging this yet because it's not being built. I've asked @DavidZemon to have a look, since he's the CI guy.

**Comment by mattcaron-token** [comment]
> I'm not merging this yet because it's not being built. I've asked @DavidZemon to have a look, since he's the CI guy.

David says it's only building `token-main`, not all feature branches.

I'm fine with this.

**Pull request merged**